src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const objOfresult = Object.assign({}, state);
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(objOfresult, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const keyToDelete of action.keysToRemove) {
+          delete objOfresult[keyToDelete];
+        }
+        break;
+
+      case 'clear':
+        for (const key in objOfresult) {
+          delete objOfresult[key];
+        }
+        break;
```

Comment:
add default case

Reply:
> add default case

what should the default case do?

Reply:
@ArturErmolenko what should the default case do?